### DISCUSSION TITLE
How to transfer adaptive mesh from an app into sub-apps

### DISCUSSION THREAD

**Discussion by QuangDom** [idaholab/moose#19600] [Q&A Meshing] [answered]
Dear Moosers,

I do not know is it possible or not, but I'm looking a solution / an example to setup a multi-app that the main app or a first sub-app will do the mesh adaptive, then transfers the newly created mesh into another sub-app(s), and then those sub-app will run without doing adaptive mesh.

Thanks in advance,
/ Dom Q.

**Comment by jiangwen84** [comment, marked as answer]
I do not think we can transfer the mesh from main to sub app. One way to do this is to transfer the adaptivity information from main app to the sub app and let the sub app perform the same adaptivity. 

Take a look at:
https://github.com/idaholab/moose/blob/next/modules/level_set/src/transfers/LevelSetMeshRefinementTransfer.C

**Reply by GiudGiud** [reply]
We can clone the mesh from main app to sub app but that is probably only done once at the beginning of the simulation.

**Reply by QuangDom** [reply]
@GiudGiud - Do you mean it can be done at the very first time of the simulation, when the mesh was not refine? 
Is there any ways to save the adaptive mesh from main-app into to file (XDR / Exodus) and the sub-apps will load it in?

**Reply by QuangDom** [reply]
Thanks @jiangwen84 . It seems a solution for me. Do you know is there any input examples?

**Reply by GiudGiud** [reply]
Yes. I doubt it clones the mesh at every time step, but i havent tried it. 
Just add `clone_master_mesh = true` in the multiapps block. If it works or doesnt work, please report back. I ll add it to the documentation

**Reply by QuangDom** [reply]
Thanks @GiudGiud - The option `clone_master_mesh = true` only work for the uniform mesh refirement that we can define from [Mesh] action. It doesn't work for the mesh adaptive that we can define on the [Adaptivity] action.

**Reply by GiudGiud** [reply]
Great thanks for looking at this. I ll add that note to the documentation. 

For your problem, @jiangwen84 's solution is the way to go. Use adaptivity in the app that knows what to refine (main app?) with an AuxVariable indicator (see https://mooseframework.inl.gov/syntax/Adaptivity/Indicators/index.html), Transfer to the other app (the one that needs the refinement but does not know where to do it) and use Adaptivity there too with this same indicator. This should provide the same degree of refinement. 

Guillaume
